src/util/serialize.cpp

```diff
@@ -120,121 +119,148 @@ std::string deSerializeString32(std::istream &is)
 }
 
 ////
-//// JSON
+//// JSON-like strings
 ////
 
 std::string serializeJsonString(const std::string &plain)
 {
-	std::ostringstream os(std::ios::binary);
-	os << "\"";
+	std::string tmp;
+
+	tmp.reserve(plain.size() + 2);
+	tmp.push_back('"');
 
 	for (char c : plain) {
 		switch (c) {
 			case '"':
-				os << "\\\"";
+				tmp.append("\\\"");
 				break;
 			case '\\':
-				os << "\\\\";
-				break;
-			case '/':
-				os << "\\/";
+				tmp.append("\\\\");
 				break;
 			case '\b':
-				os << "\\b";
+				tmp.append("\\b");
 				break;
 			case '\f':
-				os << "\\f";
+				tmp.append("\\f");
 				break;
 			case '\n':
-				os << "\\n";
+				tmp.append("\\n");
 				break;
 			case '\r':
-				os << "\\r";
+				tmp.append("\\r");
 				break;
 			case '\t':
-				os << "\\t";
+				tmp.append("\\t");
 				break;
 			default: {
 				if (c >= 32 && c <= 126) {
-					os << c;
+					tmp.push_back(c);
 				} else {
-					u32 cnum = (u8)c;
-					os << "\\u" << std::hex << std::setw(4)
-						<< std::setfill('0') << cnum;
+					// We pretend that Unicode codepoints map to bytes (they don't)
```

Comment:
Does this mean Minetest's JSON serialization has always been severely broken regarding bytestrings?